packages/appsync-modelgen-plugin/src/visitors/appsync-json-metadata-visitor.ts

```diff
@@ -107,6 +108,7 @@ export class AppSyncJSONVisitor<
   ) {
     super(schema, rawConfig, additionalConfig, defaultScalars);
     this._parsedConfig.metadataTarget = rawConfig.metadataTarget || 'javascript';
+    this._parsedConfig.codegenVersion = rawConfig.codegenVersion;
```

Comment:
This line can be moved to `appsync-visitor.ts` as it is originally defined in the super class and can be consumed by other libraries later.